pkg/cmdimpl/add_test.go

```diff
@@ -171,17 +171,75 @@ func handleGitLsRemote(arglist ...interface{}) ([]byte, []byte, error) {
 	return nil, nil, fmt.Errorf("NO!")
 }
 
-var fakeGitClient = gitfakes.FakeGit{}
+var failGitClient = gitfakes.FakeGit{
+	CloneStub: func(ctx context.Context, a, b, c string) (bool, error) {
+		fmt.Println("failing clone")
+		shims.Exit(1)
+		return false, nil
+	},
+	CommitStub: func(commit git.Commit) (string, error) {
+		fmt.Println("failing commit")
+		shims.Exit(1)
+		return "", nil
+	},
+	HeadStub: func() (string, error) {
+		fmt.Println("failing head")
+		shims.Exit(1)
+		return "", nil
+	},
+	PushStub: func(ctx context.Context) error {
+		fmt.Println("failing push")
+		shims.Exit(1)
+		return nil
+	},
+	StatusStub: func() (bool, error) {
+		fmt.Println("failing status")
+		shims.Exit(1)
+		return false, nil
+	},
+	WriteStub: func(a string, b []byte) error {
+		fmt.Println("failing write")
+		shims.Exit(1)
+		return nil
+	},
+}
+
+var ignoreGitClient = gitfakes.FakeGit{
+	CloneStub: func(ctx context.Context, a, b, c string) (bool, error) {
```

Comment:
Unless you explicitly want to see the print, there is no set all methods, it defaults to noop